src/components/tour/tour_step.tsx

```diff
@@ -164,28 +163,25 @@ export const EuiTourStep: FunctionComponent<EuiTourStepProps> = ({
     );
   }
 
-  const [hasValidAnchor, setHasValidAnchor] = useState<boolean>(false);
-  const animationFrameId = useRef<number>();
-  const anchorNode = useRef<HTMLElement | null>(null);
+  const [anchorNode, setAnchorNode] = useState<HTMLElement | null>(null);
   const [popoverPosition, setPopoverPosition] = useState<EuiPopoverPosition>();
 
   const onPositionChange = (position: EuiPopoverPosition) => {
     setPopoverPosition(position);
   };
 
   useEffect(() => {
+    let timeout: number;
     if (anchor) {
-      animationFrameId.current = window.requestAnimationFrame(() => {
-        anchorNode.current = findElementBySelectorOrRef(anchor);
-        setHasValidAnchor(anchorNode.current ? true : false);
+      timeout = window.setTimeout(() => {
+        setAnchorNode(findElementBySelectorOrRef(anchor));
       });
     }
 
     return () => {
-      animationFrameId.current &&
-        window.cancelAnimationFrame(animationFrameId.current);
+      timeout && window.clearTimeout(timeout);
     };
-  }, [anchor]);
+  }, [anchor, setAnchorNode]);
```

Comment:
IIRC state setters shouldn't be required in dependency arrays - just curious, does our linter actually complain that it's needed?
```suggestion
  }, [anchor]);
```

Reply:
Yeah it did. Setter references don't change though, so it shouldn't cause any unexpected actions here

Reply:
> Yeah it did.

That's not what I'm seeing locally - are you sure? I was able to remove it with 0 lint errors.

And yes, I know it functionally doesn't affect any actual behavior, but there's a lot of other dependency arrays where state setters are not listed as well so it's more of 1. code consistency with the rest of the repo and 2. me checking to make sure something didn't change on this in React 18

Reply:
Huh, you're right! I saw it highlighted on my editor yesterday and it usually is directly related to eslint warnings, but maybe it just didn't refresh on time.

Anyway, I'll push an update shortly :)